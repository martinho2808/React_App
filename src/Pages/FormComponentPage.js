// Functional page component
import ShoppingList from "../Components/ShoppingList";
import CurrencyConverter from "../Components/CurrencyConverter";
export default function FormCompoentPage() {
  const shopping = [
    { item: "Potato" },
    { item: "Bacon" },
    { item: "Mayonnaise" },
  ];
  return (
    <>
      <ShoppingList list={shopping} name={"Jeremy"} />
      <CurrencyConverter />
    </>
  );
}

// Class based component
// import React from "react";
// import ShoppingList from "../Components/ShoppingList";
// import CurrencyConverter from "../Components/CurrencyConverter";

// export default class FormComponentPage extends React.Component {
//   render() {
//     const shopping = [
//       { item: "Potato" },
//       { item: "Bacon" },
//       { item: "Mayonnaise" },
//     ];
//     return (
//       <div className="pageContainer">
//         <ShoppingList list={shopping} name={"Jeremy"} />
//         <CurrencyConverter />
//       </div>
//     );
//   }
// }
