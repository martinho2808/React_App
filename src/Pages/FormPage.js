import { useState, useRef } from "react";
import axios from "axios";

function FormPage() {
  const [email, setEmail] = useState("insert email here");
  const [password, setPassword] = useState("insert password");
  const [selectedFile, setSelectedFile] = useState(null);
  const [isFilePicked, setIsFilePicked] = useState(false);
  function handleFileChange(e) {
    setSelectedFile(e.target.files[0]);
    setIsFilePicked(true);
  }
  function handleFormSubmit(e) {
    e.preventDefault();
    console.log("Submission!");
    let newFormData = new FormData(e.target);
    const token = localStorage.getItem("token");
    console.log(newFormData.getAll("file"));
    console.log(newFormData.getAll("email"));
    console.log(newFormData.getAll("password"));
    axios.post(`${process.env.REACT_APP_API_SERVER}/api/form`, newFormData, {
      headers: {
        Authorization: `Bearer ${token}`,
        "content-type": "multipart/form-data",
      },
    });

    setEmail("insert email here");
    setPassword("insert password here");
    setSelectedFile(null);
    setIsFilePicked(false);
  }
  return (
    <>
      <h3>Form page test</h3>
      <form onSubmit={handleFormSubmit}>
        <label>Email: </label>
        <br />
        <input
          type="text"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          name="email"
        />
        <br />
        <label>Password: </label>
        <br />
        <input
          type="text"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          name="password"
        />
        <br />
        <label>File: </label>
        <br />
        <input type="file" onChange={handleFileChange} name="file" />
        <br />
        <input type="submit" value="submit" />
      </form>
    </>
  );
}

export default FormPage;
