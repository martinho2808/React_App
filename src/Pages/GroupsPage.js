const GroupsPage = () => {
  return (
    <>
      <h1>Welcome to the GroupsPage</h1>
      <p>We are all in special groups now....</p>
    </>
  );
};
export default GroupsPage;
