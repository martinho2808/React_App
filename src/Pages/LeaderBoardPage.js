import React from "react";
import LeaderBoard from "../Components/Exercise/LeaderBoard.js";

export default class LeaderBoardPage extends React.Component {
  render() {
    return (
      <>
        <LeaderBoard />
      </>
    );
  }
}
