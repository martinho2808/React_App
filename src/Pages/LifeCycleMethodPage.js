import React from "react";
import WhoIsInSpace from "../Components/WhoIsInSpace.js";
import Clock from "../Components/Clock";

export default class LifeCycleMethodPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      clock: false,
    };
  }

  getClock = () => {
    this.setState({
      clock: !this.state.clock,
    });
  };
  render() {
    return (
      <div className="pageContainer">
        <button onClick={this.getClock}>GetClock</button>
        {this.state.clock ? <Clock /> : "No clock"}
        <WhoIsInSpace />
      </div>
    );
  }
}
