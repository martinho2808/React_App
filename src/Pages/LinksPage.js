import LinkList from "../Components/LinkList";
export default function LinksPage() {
  <LinkList />;
}
