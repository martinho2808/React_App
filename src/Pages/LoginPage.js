import LoginForm from "../Components/LoginForm";
export default function LoginPage() {
  <LoginForm />;
}
