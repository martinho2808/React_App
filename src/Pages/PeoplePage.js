import PersonList from "../Components/PersonList";
export default function PersonPage() {
  <PersonList />;
}
