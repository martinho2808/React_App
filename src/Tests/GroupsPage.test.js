import React from "react";

import renderer from "react-test-renderer";

import { GroupsPage } from "../Pages/GroupsPage";

describe("<GroupsPage />", () => {
  it("renders complete snapshot", () => {
    const component = renderer.create(<GroupsPage />);
    let tree = component.toJSON();
    component.unmount();
    expect(tree).toMatchSnapshot();
  });
});
