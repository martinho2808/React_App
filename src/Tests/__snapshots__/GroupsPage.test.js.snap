// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupsPage /> renders complete snapshot 1`] = `
Array [
  <h1>
    Welcome to the GroupsPage
  </h1>,
  <p>
    We are all in special groups now....
  </p>,
]
`;
